tests/cypress/e2e/block.test.js

```diff
@@ -0,0 +1,54 @@
+Cypress.on( 'uncaught:exception', ( err, runnable ) => {
+	return false;
+} );
+
+describe( 'Admin can publish posts with winamp block', () => {
+	it( 'Can insert the block and publish the post', () => {
+		// login to wordpress
+		cy.login();
+
+		cy.setPermalinkStructure( '/%year%/%postname%/' );
+
+		// upload media used for the block
+		cy.uploadMedia( 'tests/cypress/fixtures/example.mp3' );
+
+		// create new page
+		cy.visit( '/wp-admin/post-new.php' );
+
+		// close the welcome guide dialog
+		//cy.get('button[aria-label="Close dialog"]').click();
+		cy.closeWelcomeGuide();
+
+		// add page title
+		cy.get( 'h1.editor-post-title' ).first().as( 'title-input' );
+		cy.get( '@title-input' ).click().type( 'Test Winamp Block' );
+
+		// insert block
+		cy.insertBlock( 'tenup/winamp-block' );
+
+		// select mp3
+		cy.get(
+			'.wp-block-tenup-winamp-block .components-button.is-tertiary'
+		).click();
+		cy.contains( '.filename div', 'example.mp3' )
+			.closest( '.thumbnail' )
+			.click();
+		cy.get( '.media-modal .media-button-select' ).click();
+
+		// publish page
+		cy.get( '.editor-post-publish-panel__toggle' ).click();
+		cy.get(
+			'.editor-post-publish-panel__header-publish-button .editor-post-publish-button__button'
+		).click();
+
+		cy.get( '.components-snackbar', { timeout: 10000 } ).should(
+			'be.visible'
+		);
+
+		// verify page contains new block
+		cy.get( 'a.components-button.components-snackbar__action' ).click();
+		cy.get( '.wp-block-audio audio' )
+			.should( 'have.attr', 'data-src' )
+			.and( 'include', 'example' );
```

Comment:
Let's use the `createPost` command here, because it is compatible with different WordPress core versions. F.e. you are using `cy.get( 'h1.editor-post-title' )` selector to get the post title input, but it doesn't work in WP 5.8 (it doesn't have a `h1` but a `div` instead).

```suggestion
		cy.createPost({
			title: 'Test Winamp Block',
			beforeSave: () => {
				cy.insertBlock( 'tenup/winamp-block' );

				// select mp3
				cy.get(
					'.wp-block-tenup-winamp-block .components-button.is-tertiary'
				).click();

				cy.get('#menu-item-browse').click();

				cy.contains( '.filename div', 'example.mp3' )
					.closest( '.thumbnail' )
					.click();
				cy.get( '.media-modal .media-button-select' ).click();
			}
		}).then( post => {
			cy.visit( `/wp-admin/post.php?post=${post.id}&action=edit` );

			// verify page contains new block
			cy.get( '.wp-block-audio audio' )
				.should( 'have.attr', 'src' )
				.and( 'include', 'example' );
		});
```

Reply:
Thanks @cadic - I find the tests are still failing at the same point. The error is:

`<h1 contenteditable="true" class="wp-block wp-block-post-title block-editor-block-list__block editor-post-title editor-post-title__input rich-text" aria-label="Add title" role="textbox" aria-multiline="true" style="white-space: pre-wrap; min-width: 1px;">...</h1>`

is being covered by another element:

<img src="https://s.w.org/images/block-editor/welcome-canvas.gif" width="312" height="240" alt="">`

Reply:
From the logs in GitHub Actions I see there is only WP minimum test fails with the error

```AssertionError: Timed out retrying after 4000ms: Expected to find element: `.components-snackbar, .components-notice.is-success`, but never found it.```

This is what I was talking about: WordPress 5.8 is unable to save post with WinAmp plugin active

Reply:
https://github.com/10up/retro-winamp-block/actions/runs/4375220604/jobs/7655657755